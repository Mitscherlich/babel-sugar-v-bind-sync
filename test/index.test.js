import { it, expect } from 'vitest'
import { transformAsync } from '@babel/core';
import transformJsx from '@vue/babel-plugin-transform-vue-jsx';
import syntaxJSX from '@babel/plugin-syntax-jsx';
import vBindSync from '..';

const code = `function createApp() {
  return (
    <div>
      <component visible_sync={this.test} foo="bar">This will be ok</component>
      <component visible_sync={this.test[1]} foo="bar">This will be ok</component>
      <component visible_sync={this.test.a} foo="bar">This will be ok</component>
      <component visible_sync={this.test.a} open_sync={this.test.b}>This will be ok</component>
      <component visible_sync={this.test} on={{click:this.test}}>This will be ok</component>
    </div>
  )
}`

it('should compiled match snapshot', async () => {
  const output = await compile(code)

  expect(output).matchSnapshot()
})

function compile(input) {
  return transformAsync(input, {
    plugins: [syntaxJSX, vBindSync, transformJsx],
  }).then(({ code }) => code)
}
