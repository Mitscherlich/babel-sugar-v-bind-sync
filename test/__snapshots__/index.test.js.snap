// Vitest Snapshot v1

exports[`should compiled match snapshot 1`] = `
"import _mergeJSXProps6 from \\"@vue/babel-helper-vue-jsx-merge-props\\";
import _mergeJSXProps5 from \\"@vue/babel-helper-vue-jsx-merge-props\\";
import _mergeJSXProps4 from \\"@vue/babel-helper-vue-jsx-merge-props\\";
import _mergeJSXProps3 from \\"@vue/babel-helper-vue-jsx-merge-props\\";
import _mergeJSXProps2 from \\"@vue/babel-helper-vue-jsx-merge-props\\";
import _mergeJSXProps from \\"@vue/babel-helper-vue-jsx-merge-props\\";

function createApp() {
  return h(\\"div\\", [h(\\"component\\", _mergeJSXProps([{
    \\"attrs\\": {
      \\"visible\\": this.test,
      \\"foo\\": \\"bar\\"
    }
  }, {
    \\"on\\": {
      \\"update:visible\\": $$val => {
        this.test = $$val;
      }
    }
  }]), [\\"This will be ok\\"]), h(\\"component\\", _mergeJSXProps2([{
    \\"attrs\\": {
      \\"visible\\": this.test[1],
      \\"foo\\": \\"bar\\"
    }
  }, {
    \\"on\\": {
      \\"update:visible\\": $$val => {
        this.test[1] = $$val;
      }
    }
  }]), [\\"This will be ok\\"]), h(\\"component\\", _mergeJSXProps3([{
    \\"attrs\\": {
      \\"visible\\": this.test.a,
      \\"foo\\": \\"bar\\"
    }
  }, {
    \\"on\\": {
      \\"update:visible\\": $$val => {
        this.test.a = $$val;
      }
    }
  }]), [\\"This will be ok\\"]), h(\\"component\\", _mergeJSXProps4([{
    \\"attrs\\": {
      \\"visible\\": this.test.a,
      \\"bool\\": true
    }
  }, {
    \\"on\\": {
      \\"update:visible\\": $$val => {
        this.test.a = $$val;
      }
    }
  }]), [\\"This will be ok\\"]), h(\\"component\\", _mergeJSXProps5([{
    \\"attrs\\": {
      \\"visible\\": this.test.a,
      \\"open\\": this.test.b
    }
  }, {
    \\"on\\": {
      \\"update:visible\\": $$val => {
        this.test.a = $$val;
      },
      \\"update:open\\": $$val => {
        this.test.b = $$val;
      }
    }
  }]), [\\"This will be ok\\"]), h(\\"component\\", _mergeJSXProps6([{
    \\"attrs\\": {
      \\"visible\\": this.test
    }
  }, {
    \\"on\\": {
      click: this.test
    }
  }, {}, {
    \\"on\\": {
      \\"update:visible\\": $$val => {
        this.test = $$val;
      }
    }
  }]), [\\"This will be ok\\"])]);
}"
`;
